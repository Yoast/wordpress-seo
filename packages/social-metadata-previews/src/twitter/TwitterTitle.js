/* External dependencies */
import styled from "styled-components";

// Used to make sure the element also has a height when empty by setting min-height equal to line-height.
const height = "18px";

const TwitterTitle = styled.p`
	line-height: ${ height };
	min-height : ${ height };
	font-size: 15px;
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
	margin-top: 0;
	margin-bottom: 2px;
	color: rgb(20, 23, 26);
	cursor: pointer;
`;

export default TwitterTitle;
