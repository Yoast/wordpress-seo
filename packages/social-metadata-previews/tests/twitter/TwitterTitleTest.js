/* External dependencies */
import React from "react";
import renderer from "react-test-renderer";

/* Internal dependencies */
import TwitterTitle from "../../src/twitter/TwitterTitle";

describe( "TwitterTitle", () => {
	it( "matches the snapshot by default", () => {
		const component = renderer.create(
			<TwitterTitle title="My Twitter Title" />
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
