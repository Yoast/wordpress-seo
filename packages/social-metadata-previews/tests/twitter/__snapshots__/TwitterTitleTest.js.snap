// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwitterTitle matches the snapshot by default 1`] = `
.c0 {
  line-height: 18px;
  min-height: 18px;
  font-size: 15px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-top: 0;
  margin-bottom: 2px;
  color: rgb(20,23,26);
  cursor: pointer;
}

<p
  className="c0"
  title="My Twitter Title"
/>
`;
